pre_commit_hooks/php_lint.sh

```diff
@@ -1,6 +1,28 @@
-#!/bin/bash
+#!/usr/bin/env bash
+
+set -o errexit
+set -o pipefail
+set -o nounset
+
+export PATH=$PATH:/usr/local/bin
+
+if ! which php-cs-fixer &>/dev/null; then
+    >&2 echo 'php-cs-fixer command not found'
+fi
+if [ `uname` = "Darwin" ]; then
+    if [ -z "$(command -v brew)" ]; then
+        exit 1
+    elif [ -z "$(command -v php-cs-fixer)" ]; then
+            brew install php-cs-fixer
+    fi
+elif [ `uname` = "Linux" ]; then
+    if [ -z "$(command -v php-cs-fixer)" ]; then
+        sudo curl -fsSL https://github.com/FriendsOfPHP/PHP-CS-Fixer/releases/download/v2.16.1/php-cs-fixer.phar -o /usr/local/bin/php-cs-fixer
+        sudo chmod 0755 /usr/local/bin/php-cs-fixer
+    fi
+fi
 
 for line in "$@"
 do
-    docker run -i --rm -v `pwd`:/opt/project gcr.io/tradesy-playground/php-linter:latest fix "$line" --rules=@PSR2 --allow-risky=no
```

Comment:
I am very curious, how does it work when you do
`docker run -i --rm -v `pwd`:/opt/project:cached gcr.io/tradesy-playground/php-linter:latest ...`
or
`docker run -i --rm -v `pwd`:/opt/project:delegated gcr.io/tradesy-playground/php-linter:latest ...`

Can you time the 3 cases? `delegated`, `cached` and `consistent` (this is the default mode, which is what you have in there).

I am not completely sure which one should be the case for this one, I am a bit more inclined for `cached` but `delegated` might provide what is needed probably.

https://docs.docker.com/docker-for-mac/osxfs-caching/

Reply:
I can. But sometimes docker just hangs for seconds at a time, when laptop gets slow, at that point you can't even commit, very frustrating.

Reply:
Also, any lag caused by docker is exacerbated by the fact that this cmd gets executed per file. That's N times based on how many files are in your commit.  It gets super painful, and devs start to uninstall pre-commit hooks because it does more damage than good.

Reply:
Actual execution and parsing is comparable.  The spin up is what is too slow.

Reply:
Using cached, delegated and without both.  The execution times jump around between .2 and .6 seconds.  WIth a few outliers.  
I think after the container spins up, file system is mounted and ready, the actual parsing/linting and fixing of files is comparable than to host system. 
The issue is with docker spinning up. Too often it gets slow. 
@nanodocumet @porkcube